### PR TITLE
Feat localrecording disclaimer

### PR DESCRIPTION
**Pull request by N7Remus**
#914

### CODE REVIEW

File: server/lib/config/config.ts [hunk 1]
```diff
@@ -1,11 +1,11 @@
 import * as fs from 'fs';
-import * as path from 'path';
+import *  as path from 'path';
 import convict from 'convict';
 import { ipaddress, url } from 'convict-format-with-validator';
 import json5 from 'json5';
 import yaml from 'yaml';
 import toml from 'toml';
-import { cpus, networkInterfaces } from 'os';
```
Comment (roman-drozd-it) [inline server/lib/config/config.ts:8]: Seems like after merging `develop` into your branch the @sd4v1d  's changes (new functionality: detecting ip) has been completely lost. Bring the `develop` version of this file back.

**Review by roman-drozd-it — CHANGES_REQUESTED** [review]

File: app/src/components/Room.js [hunk 2]
```diff
@@ -15,6 +15,7 @@ import SwipeableDrawer from '@material-ui/core/SwipeableDrawer';
 import Drawer from '@material-ui/core/Drawer';
 import Hidden from '@material-ui/core/Hidden';
 import Notifications from './Notifications/Notifications';
```
[the remark this answers is not in the document]
Comment (N7Remus) [reply app/src/components/Room.js:17]: SwipeableDrawer is still used in the Room.js, im not sure why it needs to be removed.

**Review by roman-drozd-it — CHANGES_REQUESTED** [review]
remove below lines (no used)
https://github.com/edumeet/edumeet/blob/feat-localrecording-disclaimer/server/lib/Peer.js#L8-L12
```
// Recoding STATE
const RECORDING_STOP='stop';
const RECORDING_START='start';
const RECORDING_PAUSE='pause';
const RECORDING_RESUME='resume';
```

File: app/src/components/Room.js (hunk 2 above)
[the remark this answers is not in the document]
Comment (roman-drozd-it) [reply app/src/components/Room.js]: i guess it was about this line import Notifications from './Notifications/Notifications'; (the line right before comment)

**Comment by N7Remus** [comment]
> 
 I tried to do Håvars suggestions and moved the states from the room to recordingReducer.

**Review by roman-drozd-it — APPROVED** [review]

File: app/README.md [hunk 3]
```diff
@@ -33,8 +33,9 @@ can be found here: [config.example.js](public/config/config.example.js).
...
+| simulcastProfiles | Define different encodings for various resolutions of the video. | `"object"` | ``{  "320": [    {      "scaleResolutionDownBy": 1,      "maxBitRate": 250000    }  ],  "640": [    {      "scaleResolutionDownBy": 2,      "maxBitRate": 250000    },    {      "scaleResolutionDownBy": 1,      "maxBitRate": 900000    }  ],  "1280": [    {      "scaleResolutionDownBy": 4,      "maxBitRate": 250000    },    {      "scaleResolutionDownBy": 2,      "maxBitRate": 900000    },    {      "scaleResolutionDownBy": 1,      "maxBitRate": 3000000    }  ],  "1920": [    {      "scaleResolutionDownBy": 4,      "maxBitRate": 750000    },    {      "scaleResolutionDownBy": 2,      "maxBitRate": 1500000    },    {      "scaleResolutionDownBy": 1,      "maxBitRate": 4000000    }  ],  "3840": [    {      "scaleResolutionDownBy": 4,      "maxBitRate": 1500000    },    {      "scaleResolutionDownBy": 2,      "maxBitRate": 4000000    },    {      "scaleResolutionDownBy": 1,      "maxBitRate": 10000000    }  ]}`` |
```
Comment (so010) [inline app/README.md:36]: Is there a reason you change simulcastProfiles?
I think this rolls back some testing/optimisations we made.

File: app/public/config/config.example.js [hunk 4]
```diff
@@ -83,55 +83,55 @@ var config = {
...
 			},
 			{
 				'scaleResolutionDownBy' : 2,
-				'maxBitRate'            : 500000
+				'maxBitRate'            : 900000
 			},
 			{
 				'scaleResolutionDownBy' : 1,
-				'maxBitRate'            : 1200000
+				'maxBitRate'            : 3000000
```
Comment (so010) [inline app/public/config/config.example.js:110]: Same comment as to simulcastProfiles: why?

File: app/src/RoomClient.js [hunk 5]
```diff
@@ -86,31 +86,31 @@ const PC_PROPRIETARY_CONSTRAINTS =
...
 {
 	3840 :
-		[
-			{ scaleResolutionDownBy: 12, maxBitRate: 150000 },
-			{ scaleResolutionDownBy: 6, maxBitRate: 500000 },
-			{ scaleResolutionDownBy: 1, maxBitRate: 10000000 }
-		],
+	[
+		{ scaleResolutionDownBy: 4, maxBitRate: 1500000 },
+		{ scaleResolutionDownBy: 2, maxBitRate: 4000000 },
```
Comment (so010) [inline app/src/RoomClient.js:91]: Same comment as to simulcastProfiles: why?

File: app/src/RoomClient.js [hunk 6]
```diff
@@ -680,14 +680,7 @@ export default class RoomClient
...
 
-		const { picture } = data;
-
-		let displayName;
-
-		if (typeof(data.displayName) === 'undefined' || !data.displayName)
-			displayName = '';
-		else
-			displayName = data.displayName;
+		const { displayName, picture } = data;
```
Comment (so010) [inline app/src/RoomClient.js:683]: Why do you remove the sanity check? I'm not sure what this is about can you explain?

File: app/src/config.ts [hunk 7]
```diff
@@ -162,32 +162,32 @@ This value must match exactly one of the values defined in aspectRatios.`,
...
+				{ scaleResolutionDownBy: 2, maxBitRate: 250000 },
+				{ scaleResolutionDownBy: 1, maxBitRate: 900000 }
+			],
 			320 :
-				[
-					{ scaleResolutionDownBy: 1, maxBitRate: 150000 }
-				]
+			[
+				{ scaleResolutionDownBy: 1, maxBitRate: 250000 }
+			]
```
Comment (so010) [inline app/src/config.ts:190]: Why?

**Review by so010 — CHANGES_REQUESTED** [review]

File: app/src/config.ts (hunk 7 above)
Comment (N7Remus) [reply app/src/config.ts:190]: I think some files got overwritten when i did the merge with develop.

**Pull request closed without merging**